akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala

```diff
@@ -132,20 +131,31 @@ private[akka] final class TestProbeImpl[M](name: String, system: ActorSystem[_])
     expectMessage(max.asScala, hint, obj)
 
   private def expectMessage_internal[T <: M](max: Duration, obj: T, hint: Option[String] = None): T = {
-    val o = receiveOne(max)
+    val o = receiveOne_internal(max)
     val hintOrEmptyString = hint.map(": " + _).getOrElse("")
     assert(o != null, s"timeout ($max) during expectMessage while waiting for $obj" + hintOrEmptyString)
     assert(obj == o, s"expected $obj, found $o" + hintOrEmptyString)
     o.asInstanceOf[T]
   }
 
+  override def receiveOne(): M = receiveOne(remainingOrDefault)
+
+  override def receiveOne(max: java.time.Duration): M = receiveOne(max.asScala)
+
+  def receiveOne(max: Duration): M = {
+    val o = receiveOne_internal(remainingOrDefault)
+    assert(Option(o).nonEmpty, s"Timeout ($remainingOrDefault) during receiveOne while waiting for message.")
+    assert(o.isInstanceOf[M], s"Received unexpected message type, found $o")
+    o.asInstanceOf[M]
+  }
+
   /**
    * Receive one message from the internal queue of the TestActor. If the given
    * duration is zero, the queue is polled (non-blocking).
    *
    * This method does NOT automatically scale its Duration parameter!
    */
-  private def receiveOne(max: Duration): M = {
+  private def receiveOne_internal(max: Duration): M = {
```

Comment:
Another PR was merged to master which is in conflict with this. It changed this to return Option[M]
https://github.com/akka/akka/pull/25863/files#diff-b1b4892a2377d9a513bf879eaa59fc7fR149